pkg/runner/run_context.go

```diff
@@ -643,6 +636,44 @@ func (rc *RunContext) runsOnImage(ctx context.Context) string {
 	return ""
 }
 
+func (rc *RunContext) runsOnPlatformNames(ctx context.Context) []string {
+	job := rc.Run.Job()
+
+	runsOnString, runsOnList, err := job.RunsOn()
+	if err != nil {
+		common.Logger(ctx).Errorf("'runs-on' key not defined or invalid in %s", rc.String())
+	}
+
+	platformNames := []string{}
+	if runsOnString != "" {
+		var runsOn interface{}
+		if !strings.Contains(runsOnString, "${{") || !strings.Contains(runsOnString, "}}") {
+			runsOn = runsOnString
+		} else {
+			var runsOnErr error
+			runsOn, runsOnErr = rc.ExprEval.evaluate(ctx, runsOnString, exprparser.DefaultStatusCheckNone)
+			if runsOnErr != nil {
+				common.Logger(ctx).Errorf("Unable to interpolate expression '%s': %s", runsOnString, runsOnErr)
+			}
+		}
```

Comment:
there's some overlap here with ExpressionEvaluator#Interpolate, but it seemed like sharing that would be a bigger refactor and not actually reduce any complexity. open to suggestions on better places to put this